pkg/operator/ceph/spec/spec_test.go

```diff
@@ -0,0 +1,53 @@
+/*
+Copyright 2018 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package spec
+
+import (
+	"testing"
+
+	"github.com/rook/rook/pkg/operator/k8sutil"
+	"github.com/rook/rook/pkg/operator/test"
+)
+
+func TestPodVolumes(t *testing.T) {
+	type args struct {
+		dataDirHostPath string
+	}
+	t.Run("Empty string dataDirHostPath is EmptyDir volume", func(t *testing.T) {
```

Comment:
I'm sort of on the fence about changing this. But I'm definitely not opposed to changing it if it will improve legibility.

I think a decent (but not extremely strong) reason to leave it as-is (or similar) is because the test can be called `Test<MethodName>`, and the `t.Run(...)` runs 2 sub-tests within that test. You might find it helpful to see the generated unit test framework that I was started with here:

https://github.com/rook/rook/pull/2114/files#diff-414443d7cfcbf78a7106712b7a3ca6aeR33
(Called `TestArgumentsMatchExpected` if the link breaks.)

I think I might take this opportunity to read a couple articles about unit testing in Golang to see if there are usual idiomatic patterns for this.

Reply:
If tests need some common setup, there is the "suite" pattern we are using in the integration tests to setup rook (search for the package 'github.com/stretchr/testify/suite'). We could use it for unit tests as well, but generally unit tests are small enough that it is simplest just to define new, independent tests. The pattern you're using works fine too, it's just a new pattern.

Reply:
Based on a few blogs I read, the `t.Run("test description-slash-title", fun(t *testing.T) { .. }` pattern didn't come up. I did see that there weren't multiple defined `Test...` functions to test a single function. It seems the `t.Run(...)` could be useful for nicely defined test matrices, but for something simple like this, I agree it's a little too much. I simplified it to be basically what you suggested, just with both `EmptyDir` and `HostPath` tests in the same "unit".

I appreciate your feedback. I'm still becoming familiar with golang, and I know there are going to be places where what I'm doing diverges from existing patterns in Rook and/or existing best practices in go, and I value learning those.

Reply:
sounds good, glad to talk through these questions. I have lots to learn too!